src/dishka/_adaptix/type_tools/basic_utils.py

```diff
@@ -135,3 +135,8 @@ def get_type_vars_of_parametrized(tp: TypeHint) -> VarTuple[TypeVar]:
 
 def eval_forward_ref(namespace: dict[str, Any], forward_ref: ForwardRef):
     return forward_ref._evaluate(namespace, None, recursive_guard=frozenset())
+
+
+def is_type_alias_type(tp: TypeHint) -> bool:
```

Comment:
Let's move this function back to key.py
`_adaptix` is a copy of https://github.com/reagento/adaptix, so we do not modify it. It is used to allow user to have another version of adaptix if he wants.